homeassistant/components/sensor/emoncms.py

```diff
@@ -0,0 +1,224 @@
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return self._unit_of_measurement
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the atrributes of the sensor."""
+        return {
+            ATTR_FEEDID: self._feedid,
+            ATTR_TAG: self._tag,
+            ATTR_FEEDNAME: self._feedname,
+            ATTR_SIZE: self._size,
+            ATTR_USERID: self._userid,
+            ATTR_LASTUPDATETIME: self._lastupdatetime,
+            ATTR_LASTUPDATETIMESTR: template.timestamp_local(
+                float(self._lastupdatetime)),
+        }
+
+    def update(self):
+        """Get the latest data and updates the state."""
+        if self._data is None:
+            return
+
+        found = False
+        if self._data.data is not None:
+            for elem in self._data.data:
+                elemid = get_id(self._sensorid, elem["tag"], elem["name"],
+                                elem["id"], elem["userid"])
+                if elemid == self._identifier:
+                    found = True
+                    self._lastupdatetime = elem["time"]
+                    self._size = elem["size"]
+                    self._value = elem["value"]
+                    break
+
+        if found:
+            if self._value_template is not None:
+                self._state = template.render_with_possible_json_value(
+                    self._hass, self._value_template, self._value,
+                    STATE_UNKNOWN)
+            else:
+                self._state = round(float(self._value), DECIMALS)
+
+
+# pylint: disable=too-few-public-methods
+class EmonCmsData(object):
+    """The class for handling the data retrieval."""
+
+    def __init__(self, hass, url, apikey, interval):
+        """Initialize the data object."""
+        self._apikey = apikey
+        self._url = "{}/feed/list.json".format(url)
+        self._interval = interval
+        self._hass = hass
+        self.data = None
+        self.update(dt_util.utcnow())
+
+    def update(self, now):
+        """Get the latest data."""
+        try:
+            req = requests.get(self._url, params={"apikey": self._apikey},
+                               verify=False, allow_redirects=True,
+                               timeout=5)
+        except requests.exceptions.RequestException as exception:
+            _LOGGER.error(exception)
+            return
+        else:
+            if req.status_code == 200:
+                self.data = req.json()
+            else:
+                _LOGGER.error("please verify if the specified config value "
+                              "'%s' is correct! (HTTP Status_code = %d)",
+                              CONF_URL, req.status_code)
+        finally:
+            if self.data is not None:
+                track_point_in_utc_time(self._hass,
```

Comment:
Instead of tracking this yourself, let Home Assistant polling trigger updates. 

- `EmonCmsSensor.update()` calls `EmonCmsData.update()`.
- Add `@Throttle(…)` to `EmonCmsData.update()`.

See [Bitcoin sensor](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/sensor/bitcoin.py#L178) for an example